neo4j/transaction_with_context.go

```diff
@@ -70,26 +88,23 @@ func (tx *explicitTransaction) Run(ctx context.Context, cypher string, params ma
 	}
 	stream, err := tx.conn.RunTx(ctx, tx.txHandle, db.Command{Cypher: cypher, Params: params, FetchSize: tx.fetchSize})
 	if err != nil {
-		tx.err = err
-		tx.runFailed = true
+		tx.txState.onError(err)
 		tx.onClosed(tx)
-		return nil, errorutil.WrapError(tx.err)
+		return nil, errorutil.WrapError(tx.txState.err)
 	}
 	// no result consumption hook here since bookmarks are sent after commit, not after pulling results
-	return newResultWithContext(tx.conn, stream, cypher, params, nil), nil
+	result := newResultWithContext(tx.conn, stream, cypher, params, tx.txState, nil)
+	tx.txState.resultErrorHandlers = append(tx.txState.resultErrorHandlers, result.errorHandler)
+	return result, nil
 }
 
 func (tx *explicitTransaction) Commit(ctx context.Context) error {
-	if tx.runFailed {
-		tx.runFailed = false
-		return tx.err
-	}
-	if tx.conn == nil {
+	if tx.conn == nil || tx.conn.HasFailed() {
```

Comment:
`tx.conn.HasFailed()` was missing here causing `test_should_prevent_commit_after_tx_termination` to fail.

Reply:
should you also check `|| !tx.conn.IsAlive()` as is done in `Rollback()`?

Reply:
I think so, I've re-ran Testkit with this change and all seems good 👍.